plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -1,13 +1,247 @@
+   *
+   * @param facetRuntime the facet runtime; runtime should not be null
+   * @return true is <code>facetRuntime</code> is an App Engine Standard runtime and false otherwise
+   */
+  public static boolean isAppEngineRuntime(IRuntime facetRuntime) {
+    Preconditions.checkNotNull(facetRuntime, "runtime is null");
+
+    org.eclipse.wst.server.core.IRuntime serverRuntime = FacetUtil.getRuntime(facetRuntime);
+    if (serverRuntime != null) {
+      IRuntimeType runtimeType = serverRuntime.getRuntimeType();
+      return runtimeType.getId().equals(DEFAULT_RUNTIME_ID);
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Returns true is <code>serverRuntime</code> is an App Engine Standard runtime and false otherwise
+   *
+   * @param serverRuntime the server runtime, runtime should not be null
+   * @return true if <code>serverRuntime</code> is an App Engine Standard runtime and false otherwise
+   */
+  public static boolean isAppEngineRuntime(org.eclipse.wst.server.core.IRuntime serverRuntime) {
+    Preconditions.checkNotNull(serverRuntime, "runtime is null");
+    IRuntimeType runtimeType = serverRuntime.getRuntimeType();
+    return runtimeType.getId().equals(DEFAULT_RUNTIME_ID);
+  }
+
+  /**
+   * Checks to see if <code>facetedProject</code> has the App Engine facet installed. If not, it installs
+   * the App Engine facet.
+   *
+   * @param facetedProject the faceted project receiving the App Engine facet
+   * @param monitor the progress monitor
+   * @throws CoreException if anything goes wrong during install
+   */
+  public static void installAppEngineFacet(IFacetedProject facetedProject, boolean installDependentFacets, IProgressMonitor monitor)
+      throws CoreException {
+    // Install required App Engine facets i.e. Java 1.7 and Dynamic Web Module 2.5
+    if (installDependentFacets) {
+      installJavaFacet(facetedProject, monitor);
+      installWebFacet(facetedProject, monitor);
+    }
+
+    IProjectFacet appEngineFacet = ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID);
+    IProjectFacetVersion appEngineFacetVersion = appEngineFacet.getVersion(AppEngineStandardFacet.VERSION);
+
+    if (!facetedProject.hasProjectFacet(appEngineFacet)) {
+      facetedProject.installProjectFacet(appEngineFacetVersion, null /* config */, monitor);
+    }
+  }
+
+  /**
+   * If App Engine runtimes exist in the workspace, add them to the list of targeted runtimes
+   * of <code>project</code>, otherwise create a new App Engine runtime and add it to the list
+   * of targeted runtimes.
+   *
+   * @param project the faceted project receiving the App Engine runtime(s)
+   * @param force true if all runtime instances should be added to the <code>project</code> even if targeted list of
+   *  <code>project</code> already includes App Engine runtime instances and false otherwise
+   * @param monitor the progress monitor
+   * @throws CoreException if method fails for any reason
```

Comment:
I copied this from one of the javadocs of the functions that throw CoreException, maybe the IFacetedProject. It is vague. Will update it.